lib/sandbox/pmapi.js

```diff
@@ -269,7 +269,28 @@ function Postman (execution, onRequest, onSkipRequest, onAssertion, cookieStore,
              * @excludeFromTestScript
              * @instance
              */
-            skipRequest: onSkipRequest
+            skipRequest: onSkipRequest,
+
+            /**
+             * @interface ExecutionLocation
+             * @extends Array<string>
+             */
+
+            /**
+             * The path of the current request.
+             *
+             * @type {ExecutionLocation} - current execution path
+             * @instance
+             */
+            location: _assignDefinedReadonly(execution.legacy._itemPath || [], /** @lends ExecutionLocation */ {
```

Comment:
`execution.legacy` is not the best way to accept this context. Need to expose a better interface for consumers to pass down all this information.